rtc.go

```diff
@@ -135,31 +145,35 @@ type RTC struct {
 	//cache datachannel api operation before dr.OnOpen
 	apiQueue []Call
 
-	client rtc.RTCClient
-	stream rtc.RTC_SignalClient
+	signaller Signaller
 
 	ctx        context.Context
 	cancel     context.CancelFunc
 	handleOnce sync.Once
 	sync.Mutex
 }
 
-func NewRTC(connector *Connector, config ...RTCConfig) *RTC {
+func NewRTC(config ...RTCConfig) *RTC {
 	r := &RTC{
-		connector: connector,
+		notify: make(chan struct{}),
 	}
+	r.ctx, r.cancel = context.WithCancel(context.Background())
 
 	if len(config) > 0 {
 		r.config = &config[0]
 	}
 
-	r.connector.RegisterService(r)
+	return r
+}
+
+func (r *RTC) Start(signaller Signaller) {
```

Comment:
Is it possible to avoid this `.Start()` call or otherwise make the dependencies linear? It seems very easy to forget to call this function, not to mention if I understand correctly this would break everyone silently instead of catching a compilation error.

As an idea, perhaps instead of making a signaller required to construct separately, it could be passed in the config instead?

Reply:
That, is doing something like `NewRTC(WithSignaller(...))` following the same pattern we use elsewhere in pion for constructor options. I think it would be reasonable to update the config to require `NewRTC(WithConfig(...), WithSignaller(...))`, for example.

Reply:
How about a just second constructor 
```
func NewRTC(connector *Connector, config ...RTCConfig) (*RTC, error){
	r := &RTC{
		notify: make(chan struct{}),
	}
	r.ctx, r.cancel = context.WithCancel(context.Background())

	if len(config) > 0 {
		r.config = &config[0]
	}

	signaller, err := connector.Signal(r)
	r.start(signaller)
	return r, err
}

func NewRTCWithSignaller(signaller Signaller, config ...RTCConfig) *RTC {
	r := &RTC{
		notify: make(chan struct{}),
	}
	r.ctx, r.cancel = context.WithCancel(context.Background())

	if len(config) > 0 {
		r.config = &config[0]
	}

	r.start(signaller)
	return r
}
```

This way all currently implementationations only change by NewRTC now returning an error as well (which they'll know at compile).

Reply:
@kevmo314 what do you think about this approach? 
The problem requiring Start is that RTC depends on connector and connector requires RTC to start signalling. 

If you're happy with this approach I'll clean up before review.

Reply:
That works too :)